azure/scope/machine_test.go

```diff
@@ -223,3 +229,829 @@ func TestMachineScope_ProviderID(t *testing.T) {
+							OSType: "Windows",
+						},
+					},
+				},
+				ClusterScoper: &ClusterScope{
+					AzureClients: AzureClients{
+						EnvironmentSettings: auth.EnvironmentSettings{
+							Environment: autorestazure.Environment{
+								Name: autorestazure.PublicCloud.Name,
+							},
+						},
+					},
+				},
+			},
+			want: []azure.VMExtensionSpec{},
+		},
+		{
+			name: "If OS type is Linux and cloud is not AzurePublicCloud, it returns empty",
+			machineScope: MachineScope{
+				Machine: &clusterv1.Machine{},
+				AzureMachine: &infrav1.AzureMachine{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "machine-name",
+					},
+					Spec: infrav1.AzureMachineSpec{
+						OSDisk: infrav1.OSDisk{
+							OSType: "Windows",
+						},
+					},
+				},
+				ClusterScoper: &ClusterScope{
+					AzureClients: AzureClients{
+						EnvironmentSettings: auth.EnvironmentSettings{
+							Environment: autorestazure.Environment{
+								Name: autorestazure.USGovernmentCloud.Name,
+							},
+						},
+					},
+				},
+			},
+			want: []azure.VMExtensionSpec{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.machineScope.VMExtensionSpecs(); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("VMExtensionSpecs() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestMachineScope_Subnet(t *testing.T) {
+	tests := []struct {
+		name         string
+		machineScope MachineScope
+		want         infrav1.SubnetSpec
+	}{
+		{
+			name: "returns empty if no subnet is found at cluster scope",
+			machineScope: MachineScope{
+				AzureMachine: &infrav1.AzureMachine{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "machine-name",
+					},
+					Spec: infrav1.AzureMachineSpec{
+						SubnetName: "machine-name-subnet",
+					},
+				},
+				ClusterScoper: &ClusterScope{
+					AzureCluster: &infrav1.AzureCluster{
+						Spec: infrav1.AzureClusterSpec{
+							NetworkSpec: infrav1.NetworkSpec{
+								Subnets: []infrav1.SubnetSpec{},
+							},
+						},
+					},
+				},
+			},
+			want: infrav1.SubnetSpec{},
+		},
+		{
+			name: "returns the machine subnet name if the same is present in the cluster scope",
+			machineScope: MachineScope{
+				AzureMachine: &infrav1.AzureMachine{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "machine-name",
+					},
+					Spec: infrav1.AzureMachineSpec{
+						SubnetName: "machine-name-subnet",
+					},
+				},
+				ClusterScoper: &ClusterScope{
+					AzureCluster: &infrav1.AzureCluster{
+						Spec: infrav1.AzureClusterSpec{
+							NetworkSpec: infrav1.NetworkSpec{
+								Subnets: []infrav1.SubnetSpec{
+									{
+										Name: "machine-name-subnet",
+									},
+									{
+										Name: "another-machine-name-subnet",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			want: infrav1.SubnetSpec{
+				Name: "machine-name-subnet",
+			},
+		},
+		{
+			name: "returns empty if machine subnet name is not present in the cluster scope",
```

Comment:
This test spec looks the same as the one above it, but I think you meant to have it return an empty `SubnetSpec`.

Reply:
Yes, thank you. That is a slip from me. Will fix this.

Reply:
fixed the test spec